### PR TITLE
feat: support setting secrets for renovate

### PR DESCRIPTION
**Pull request by gmariotti**
Hi, I created this PR to make it possible to use k8s/Secret to pass sensitive data to Renovate. I'm no Helm expert, I probably missed something for sure 😅

### CODE REVIEW

**Comment by gmariotti** [comment]
The tests expect that I bump the version of the chart but the version seems to be bounded to the version of renovate. What should I do?

**Comment by JamieMagee** [comment]
@gmariotti Sorry I've missed this PR. I'll find some time today to review it.

We decided to pin the helm chart version to the renovate version to make auto updates easier. I think we should be able to add a build version, and still stay compliant with semver. Something like `19.239.11+1`

File: charts/renovate/templates/secret.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+kind: Secret
+metadata:
+  name: {{ template "renovate.fullname" . }}-secret
+  labels:
+    app: {{ template "renovate.name" . }}
+    chart: {{ template "renovate.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+type: Opaque
+data:
```
Comment (JamieMagee) [inline charts/renovate/templates/secret.yaml:12]: I would simplify this to `stringData` and drop the `b64enc`

Comment (gmariotti) [reply charts/renovate/templates/secret.yaml:12]: I will fix it, thanks 🙏

**Comment by gmariotti** [comment]
What about documentation? Is there anything I need to do?

**Comment by JamieMagee** [comment]
Yeah, can you add an example to the `README.md` in the root of the repo, as well as the `README.md.gotmpl` and `README.md` in the `charts/renovate` directory.

And don't worry about merging, I'll figure it out 😅

File: charts/renovate/templates/secret.yaml (hunk 1 above)
Comment (gmariotti) [reply charts/renovate/templates/secret.yaml:12]: I didn't find any `stringData` function, could you point me to it?

**Comment by gmariotti** [comment]
@viceice @JamieMagee rebased with master and addressed some of the comments, but I'm still having a problem with the version number (probably because of the commit prefix used), any suggestion?

File: charts/renovate/Chart.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 apiVersion: v1
-appVersion: '22.25.6'
+appVersion: '22.25.6+1'
 description: Universal dependency update tool that fits into your workflows.
 name: renovate
 version: '22.25.6'
```
Comment (viceice) [inline charts/renovate/Chart.yaml:5]:
```suggestion
version: '22.25.6+1'
```

File: charts/renovate/Chart.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 apiVersion: v1
-appVersion: '22.25.6'
+appVersion: '22.25.6+1'
```
Comment (viceice) [inline charts/renovate/Chart.yaml:2]:
```suggestion
appVersion: '22.25.6'
```

**Comment by viceice** [comment]
Should be released with next renovate update

**Comment by viceice** [comment]
v23 on the way: renovatebot/docker-renovate-full#59

**Pull request merged**